src/ert/_c_wrappers/job_queue/workflow_job.py

```diff
@@ -80,36 +136,137 @@ def _make_arg_types_list(
             return []
 
     @classmethod
-    def fromFile(cls, config_file, name=None):
-        if os.path.isfile(config_file) and os.access(config_file, os.R_OK):
-            if name is None:
+    def _file_to_workflow_args(cls, config_file, name=None, use_new_parser=True):
+        try:
+            if not name:
                 name = os.path.basename(config_file)
 
-            content = _config_parser.parse(config_file)
+            if use_new_parser:
+                new_content_dict = new_workflow_job_parser(config_file)
 
-            def optional_get(key):
-                return content.getValue(key) if content.hasKey(key) else None
+                new_types_list = cls._make_arg_types_list_new(new_content_dict)
+                return (
+                    name,
+                    new_content_dict.get("INTERNAL"),
+                    new_content_dict.get("MIN_ARG"),
+                    new_content_dict.get("MAX_ARG"),
+                    new_types_list,
+                    new_content_dict.get("EXECUTABLE"),
+                    str(new_content_dict.get("SCRIPT"))
+                    if "SCRIPT" in new_content_dict
+                    else None,
+                ), None
+            else:
+                old_content = _config_parser.parse(config_file)
 
-            max_arg = optional_get("MAX_ARG")
+                def optional_get(key):
+                    return (
+                        old_content.getValue(key) if old_content.hasKey(key) else None
+                    )
+
+                max_arg = optional_get("MAX_ARG")
+
+                old_arg_types_list = cls._make_arg_types_list(old_content, max_arg)
+                return (
+                    name,
+                    optional_get("INTERNAL"),
+                    optional_get("MIN_ARG"),
+                    max_arg,
+                    old_arg_types_list,
+                    optional_get("EXECUTABLE"),
+                    optional_get("SCRIPT"),
+                ), None
+        except ConfigValidationError as err:
```

Comment:
This is not the only possible exception that may be raised. Its probably better to use the approach taken ert.gui.main where any exception raised from the parser we are only comparing against is catched and logged.